### DISCUSSION TITLE
Error when evaluating SSR module, but the real error is not displayed

### DISCUSSION THREAD

**Discussion by loxs** [vikejs/vike#743] [Discussion] [answered]
```
4:28:01 PM [vite] Error when evaluating SSR module ... form-data/lib/form_data.js:
4:28:01 PM [vite] Error when evaluating SSR module ... axios/lib/platform/node/classes/FormData.js:
4:28:01 PM [vite] Error when evaluating SSR module ... node_modules/axios/lib/helpers/toFormData.js:
4:28:01 PM [vite] Error when evaluating SSR module ... node_modules/axios/lib/helpers/AxiosURLSearchParams.js:
4:28:01 PM [vite] Error when evaluating SSR module ... node_modules/axios/lib/helpers/buildURL.js:
4:28:01 PM [vite] Error when evaluating SSR module ... node_modules/axios/lib/core/Axios.js:
4:28:01 PM [vite] Error when evaluating SSR module ... node_modules/axios/lib/axios.js:
4:28:01 PM [vite] Error when evaluating SSR module ... node_modules/axios/index.js:
4:28:01 PM [vite] Error when evaluating SSR module ...
4:28:01 PM [vite] Error when evaluating SSR module ...
4:28:01 PM [vite] Error when evaluating SSR module /pages/myPage.page.tsx:
```

Ignore the actual error (I have removed parts to anonymize it), it's not important. My point is that I don't get any useful info on what is the actual problem. 
I am probably missing something very basic, but I didn't find anything useful in the docs.

I have just started my app via the React + TS template and I haven't done any meaningful changes to it. Just tried writing a page. Whenever there is a problem, this is the only error output I'm getting in the console (or anywhere really).

What should I do in order to have a better experience?

**Comment by brillout** [comment]
I agree the error should be more helpful. Would you mind re-creating this ticket as a [Vite bug](https://github.com/vitejs/vite/issues/new?assignees=&labels=pending+triage&template=bug_report.yml)?

**Comment by loxs** [comment]
Huh, I can't seem to be able reproduce it in a fresh project. It does have proper error messages.
I see some commits improving error messages over the past days (which is after I started my project). Is it possible that this got fixed?

**Reply by brillout** [reply]
I actually also got these non-helpful errors yesterday. I did `$ npm init vite-plugin-ssr@latest` then:

```diff
  // vite.config.js
  
  import react from '@vitejs/plugin-react'
  import ssr from 'vite-plugin-ssr/plugin'
  
  export default {
    plugins: [react(), ssr()],
+   ssr: {
+     noExternal: true
+   }
  }
```

Feel free to CC me in your Vite bug report.

**Comment by loxs** [comment, marked as answer]
https://github.com/vitejs/vite/issues/12631

**Reply by brillout** [reply]
Workardound released at `0.4.106`.
